### PR TITLE
[noticket] Fix suggest url generation for search form

### PR DESCRIPTION
**Pull request by lx-wnk**
### 1. Why is this change necessary?
If someone needs to customize the search form only the search url would adjusted

### 2. What does this change do, exactly?
Switch from using the input value to using the whole form data

### 3. Describe each step to reproduce the issue or behaviour.
Try to add another input field to the search form and use it in the suggest call

### 4. Please link to the relevant issues (if any).
///

### 5. Checklist

- [ ] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [x] I have created a [changelog file](https://github.com/shopware/platform/blob/master/adr/2020-08-03-Implement-New-Changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [x] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: changelog/_unreleased/2020-09-14-Adjust-suggest-form-behaviour.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+ ---
+ title:              Adjust suggest form behaviour
+ issue:              
+ author:             Alexander Wink
+ author_email:       a.wink@kellerkinder.de
+ author_github:      @jinnoflife
+ ---
+ # Storefront
+ * Switch from predefined input selection to FormData-handling
```
Comment (PaddyS) [inline changelog/_unreleased/2020-09-14-Adjust-suggest-form-behaviour.md]: Please use `Created`, `Added`, `Removed` or `Deprecated` as keyword here :-)

File: changelog/_unreleased/2020-09-14-Adjust-suggest-form-behaviour.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+ ---
+ title:              Adjust suggest form behaviour
+ issue:
```
Comment (mitelg) [inline changelog/_unreleased/2020-09-14-Adjust-suggest-form-behaviour.md]: remove empty fields here

Comment (lx-wnk) [reply changelog/_unreleased/2020-09-14-Adjust-suggest-form-behaviour.md]: done

File: changelog/_unreleased/2020-09-14-Adjust-suggest-form-behaviour.md (hunk 1 above)
Comment (lx-wnk) [reply changelog/_unreleased/2020-09-14-Adjust-suggest-form-behaviour.md]: Done

File: src/Storefront/Resources/app/storefront/src/plugin/header/search-widget.plugin.js [hunk 3]
```diff
@@ -85,33 +85,42 @@ export default class SearchWidgetPlugin extends Plugin {
...
+                encodedUri += '&';
+            }
+
+            encodedUri += `${fieldName}=` + encodeURIComponent(fieldValue);
+        }
+
+        this._suggest(encodedUri);
 
-        this.$emitter.publish('handleInputEvent', { value });
+        this.$emitter.publish('handleInputEvent', { formData });
```
Comment (mitelg) [inline src/Storefront/Resources/app/storefront/src/plugin/header/search-widget.plugin.js]: isn't this breaking if the event is suddenly emitting other data?

Comment (lx-wnk) [reply src/Storefront/Resources/app/storefront/src/plugin/header/search-widget.plugin.js]: Might be.
Would you prefer another Event with the whole `FormData`  and pass for the original one only the input value?

Comment (mitelg) [reply src/Storefront/Resources/app/storefront/src/plugin/header/search-widget.plugin.js]: instead of replacing it, maybe it would be enough to add it as second parameter?

Comment (lx-wnk) [reply src/Storefront/Resources/app/storefront/src/plugin/header/search-widget.plugin.js]: Great Idea. I'll adjust that :)

Comment (klarstil) [reply src/Storefront/Resources/app/storefront/src/plugin/header/search-widget.plugin.js]: Yes, please add a second key to the object otherwise you would break the current public API of the search widget and therefore that would end up with the consequence that your PR will be part of the next major instead of the next minor versio.

Comment (lx-wnk) [reply src/Storefront/Resources/app/storefront/src/plugin/header/search-widget.plugin.js]: It's done :) 

Sorry for the late adjustment

**Comment by J-Rahe** [comment]
Hey @jinnoflife could you add an test please?

**Comment by lx-wnk** [comment]
@J-Rahe What kind of test do you need?
The test at `platform/src/Storefront/Resources/app/storefront/test/e2e/cypress/integration/search/frontend-search.spec.js` should cover the search functionality IMO.

File: src/Storefront/Resources/app/storefront/src/plugin/header/search-widget.plugin.js [hunk 4]
```diff
@@ -85,33 +85,43 @@ export default class SearchWidgetPlugin extends Plugin {
...
+                encodedUri += '&';
+            }
+
+            encodedUri += `${fieldName}=` + encodeURIComponent(fieldValue);
+        }
+
+        this._suggest(encodedUri);
 
-        this.$emitter.publish('handleInputEvent', { value });
+        this.$emitter.publish('handleInputEvent', { inputValue, formData });
```
Comment (J-Rahe) [inline src/Storefront/Resources/app/storefront/src/plugin/header/search-widget.plugin.js:109]: this might brake plugins due to another name. You should be able to just write value:inputValue
this should also be documented (changelog.md -> handleInputEvent now emits... sth. along that way)

File: src/Storefront/Resources/app/storefront/src/plugin/header/search-widget.plugin.js [hunk 5]
```diff
@@ -85,33 +85,43 @@ export default class SearchWidgetPlugin extends Plugin {
...
     }
 
     /**
      * Process the AJAX suggest and show results
      * @param {string} value
```
Comment (J-Rahe) [inline src/Storefront/Resources/app/storefront/src/plugin/header/search-widget.plugin.js:114]: value -> encodedUri

**Comment by J-Rahe** [comment]
@jinnoflife would be great if you could take a look at the suggestions. in case of an test: there is already the 
search-widget.plugin.test.js if you could extend that one to make sure the submitted data stays in the same format and is available with the same variable name,( and maybe add some(more) edge cases) that would be nice.

**Comment by J-Rahe** [comment]
This PR was closed due to inactivity. If this change is still important to you, feel free to create a new pull request.
For more information about our contribution guidelines, see https://docs.shopware.com/en/shopware-platform-dev-en/contribution/contribution-guideline
